shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/IntentHandler.java

```diff
@@ -0,0 +1,176 @@
+            String mimeType = resolver.getType(uri);
+            String extension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
+            String filename = String.format("%s.%s", UUID.randomUUID().toString(), extension);
+            File file = new File(reactContext.getCacheDir(), filename);
+            try {
+                InputStream istream = resolver.openInputStream(uri);
+                OutputStream ostream = new FileOutputStream(file);
+
+                byte[] buf = new byte[64 * 1024];
+                int len;
+                while ((len = istream.read(buf)) != -1) {
+                    ostream.write(buf, 0, len);
+                }
+                filePath = file.getPath();
+            } catch (IOException ex) {
+                Log.w(TAG, "error writing shared file " + uri.toString());
+            }
+        } else {
+            filePath = uri.getPath();
+        }
+
+        if (filePath == null) return;
+
+        if (reactContext == null) return;
+
+        WritableMap evt = Arguments.createMap();
+        evt.putString("path", filePath);
+
+        DeviceEventManagerModule.RCTDeviceEventEmitter emitter = reactContext
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class);
+        if (emitter != null) {
+            emitter.emit("onShareData", evt);
+        }
+    }
+
+    private void handleSendIntentMultipleStreams(Intent intent) {
+        ArrayList<Uri> uris = intent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+        if (uris == null) return;
+
+        // TODO: do something with the intent streams.
+    }
+
+    private void handleSendIntentText(Intent intent) {
+        String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
+        if (sharedText == null) return;
+
+        if (reactContext == null) return;
+
+        WritableMap evt = Arguments.createMap();
+        evt.putString("text", sharedText);
+        DeviceEventManagerModule.RCTDeviceEventEmitter emitter = reactContext
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class);
+        if (emitter != null) {
+            emitter.emit("onShareText", evt);
+        }
+    }
+
+    public void handleIntent(final Intent intent) {
+        if (intent == null) return;
+
+        if (reactContext == null) {
+            NativeLogger.warn("react context not ready");
+            return;
+        }
+
+        handleNotificationIntent(intent);
+        handleSendIntentText(intent);
+        handleSendIntentStream(intent);
+        handleSendIntentMultipleStreams(intent);
+    }
+
+    public IntentHandler(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+
+        reactContext.addLifecycleEventListener(new LifecycleEventListener() {
+            @Override
+            public void onHostResume() {
+                handleIntent(getCurrentActivity().getIntent());
```

Comment:
Do you need this and this one https://github.com/keybase/client/pull/16380/files#diff-39fa276eb992da2acfb5a28bc9ea4d63R102 ?

Would it work if you just used this one?

Reply:
Yup it works, thanks.